src/cider/exec/nextgen/context/RuntimeContext.cpp

```diff
@@ -49,4 +55,61 @@ void RuntimeContext::instantiate(const CiderAllocatorPtr& allocator) {
 
   string_heap_ptr_ = std::make_shared<StringHeap>(allocator);
 }
+
+void allocateBatchMem(ArrowArray* array,
+                      int64_t length,
+                      bool is_struct = true,
+                      int64_t value_size = 0) {
```

Comment:
This method will be refactored inevitably one day😂

Reply:
Just to simplify the code, the size of the value cannot be determined, thus it can't be recursive